packages/lesswrong/components/walledGarden/GardenCodesItem.tsx

```diff
@@ -20,47 +40,90 @@ const styles = theme => ({
+  trailingIcons: {
     color: theme.palette.grey[700],
-    height: 16,
-    width: 16,
-    cursor: "pointer",
-    opacity: .35,
+    width: 48,
     marginLeft: 8,
-    position: "relative",
-    top: 1,
-    '&:hover': {
-      opacity: 1
-    }
-  }
+    display: "flex",
+    justifyContent: "space-between"
+  },
+  linkIcon: {
+    ...iconStyling,
+    opacity: .5,
+    transform: 'rotate(-45deg)',
+  },
+  fbIcon: {
+    ...iconStyling,
+    color: theme.palette.grey[700],
+    height: 12,
+    width: 12,
+    opacity: .5,
+    marginBottom: 2
+  },
+  editIcon: {
+    ...iconStyling,
+    // color: theme.palette.grey[700],
+    opacity: .35,
+  },
 })
 
 export const GardenCodesItem = ({classes, gardenCode}:{
   classes:ClassesType,
   gardenCode: GardenCodeFragment
 }) => {
   const { GardenCodesEditForm, LWTooltip, ContentItemBody } = Components
+  const currentUser = useCurrentUser();
   const [editing, setEditing] = useState(false)
+  const { flash } = useMessages();
   if (editing) {
     return <GardenCodesEditForm gardenCodeId={gardenCode._id} cancelCallback={()=> setEditing(false)}   />
   }
-  const title = <span className={classes.eventName}>{gardenCode.title}</span>
+  // const title = <span cla>{gardenCode.title}</span>
+  const inviteLink = `http://garden.lesswrong.com?code=${gardenCode.code}&event=${gardenCode.slug}`
+  
   return <div className={classes.root}>
-      {gardenCode.contents?.htmlHighlight ? <span className={classes.eventName}><LWTooltip title={<span className={classes.highlight}>
-        <ContentItemBody
-          dangerouslySetInnerHTML={{__html: gardenCode.contents.htmlHighlight }}
-          description={`garden-code-${gardenCode._id}`}
-        />
-      </span>}>
-        {title}
-      </LWTooltip></span>
-        : title
+    <span className={classes.eventName}>
+      {gardenCode.type=='private' && <LWTooltip title="Your personal event" placement="right">
+        <PersonIcon className={classes.personalIcon}/>
+      </LWTooltip>}
+      {gardenCode.contents?.htmlHighlight ? <span><LWTooltip title={<span className={classes.highlight}>
+          <ContentItemBody
+            dangerouslySetInnerHTML={{__html: gardenCode.contents.htmlHighlight }}
+            description={`garden-code-${gardenCode._id}`}
+          />
+        </span>}>
+          {gardenCode.title}
+        </LWTooltip>
+      </span>
+        : gardenCode.title
       }
-      <div className={classes.eventTime}>
-        {eventFormat(gardenCode.startTime)}
-      </div>
-      <CreateIcon className={classes.icon} onClick={() => setEditing(true)}/>
+    </span>
+    <div className={classes.eventTime}>
+      {eventFormat(gardenCode.startTime)}
+    </div>
+    <div className={classes.trailingIcons}>
+      <LWTooltip title="Click to copy invite link" placement="right">
+                                                                      <CopyToClipboard text={inviteLink} onCopy={()=>flash({messageString:"Invite link copied!"})}>
```

Comment:
What happened with the indents here?

Reply:
Whoa, huh. Someone fell asleep on tab or spacebar.